lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -453,9 +453,25 @@ sub show ($self) {
     my $job_id = int($self->stash('jobid'));
     my $details = $self->stash('details') || 0;
     my $check_assets = !!$self->param('check_assets');
-    my $job = $self->schema->resultset('Jobs')->find($job_id, {prefetch => 'settings'});
-    return $self->reply->not_found unless $job;
+    my $follow = $self->param('follow');    # follow job clones and report most recent result for given id
+
+    my @cloned_as;
+    my $job;
+    while (1) {
+        return unless $job = $self->find_job_or_render_not_found($job_id, {prefetch => 'settings'});
+        if ($follow && $job->clone_id) {
+            push(@cloned_as, $job->clone_id);
+            $job_id = $job->clone_id;
+            next;
+        }
+        last;
+    }
     $job = $job->to_hash(assets => 1, check_assets => $check_assets, deps => 1, details => $details, parent_group => 1);
+    if ($job_id != $self->stash('jobid')) {
+        # if follow=1 return cloned job with original id
+        # but indicate clone chain in cloned_as
+        $job->{cloned_as} = \@cloned_as;
+    }
```

Comment:
I'd avoid the repeated querying of the job ID from the stash and write it as one-liner:

```suggestion
    $job->{cloned_as} = \@cloned_as if @cloned_as;
```

There's probably also no need for the comment.

In this short form the duplication you have in the other function is probably even acceptable.